lib/ansible/modules/cloud/vmware/vmware_local_user_manager.py

```diff
@@ -111,71 +144,243 @@ def process_state(self):
+        )
         return user_account
 
     def create_account_spec(self):
+        """Create account specification"""
         account_spec = vim.host.LocalAccountManager.AccountSpecification()
         account_spec.id = self.local_user_name
         account_spec.password = self.local_user_password
         account_spec.description = self.local_user_description
         return account_spec
 
-    def state_create_user(self):
-        account_spec = self.create_account_spec()
-
+    def set_permissions(self, user, role_id):
+        """Set specified permissions on an entity"""
+        permission_spec = vim.AuthorizationManager.Permission()
+        permission_spec.principal = user
+        permission_spec.group = False
+        permission_spec.roleId = role_id
+        permission_spec.propagate = self.propagate
         try:
-            self.content.accountManager.CreateUser(account_spec)
-            self.module.exit_json(changed=True)
-        except vmodl.RuntimeFault as runtime_fault:
-            self.module.fail_json(msg=runtime_fault.msg)
-        except vmodl.MethodFault as method_fault:
-            self.module.fail_json(msg=method_fault.msg)
+            self.content.authorizationManager.SetEntityPermissions(
+                entity=self.content.rootFolder,
+                permission=[permission_spec]
+            )
+        except vim.fault.UserNotFound as user_not_found:
+            self.module.fail_json(msg="The specified user or group does not exist : %s" % to_native(user_not_found))
+        except vim.fault.NotFound as not_found:
+            self.module.fail_json(msg="The role ID isn't valid : %s" % to_native(not_found))
+        except vim.fault.AuthMinimumAdminPermission as admin_permission:
+            self.module.fail_json(
+                msg="This change would leave the system with no Administrator permission on the root node, "
+                "or it would grant further permission to a user or group who already has Administrator permission "
+                "on the root node : %s" % to_native(admin_permission)
+            )
+        except vmodl.fault.ManagedObjectNotFound as object_not_found:
+            self.module.fail_json(msg="The specified entity doesn't exist : %s" % to_native(object_not_found))
+        except vmodl.fault.InvalidArgument as invalid_argument:
+            self.module.fail_json(
+                msg="The new role ID is the View or Anonymous role, "
+                "or the entity does not support assigning permissions : %s" % to_native(invalid_argument)
+            )
+        except vim.fault.NoPermission as no_permission:
+            self.module.fail_json(msg="No priviledge to set permission : %s" % to_native(no_permission))
+
+    @staticmethod
+    def get_role_id(role_name, role_list):
+        """Get role ID from role name"""
+        role_id = None
+        # Default roles
+        if role_name == 'NoAccess':
```

Comment:
role_name can be lower case as well. For example, user can provide admin for `Admin` value, then this function will return None.